extensions-core/druid-basic-security/src/main/java/io/druid/security/basic/authorization/BasicRoleBasedAuthorizer.java

```diff
@@ -0,0 +1,112 @@
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.fasterxml.jackson.annotation.JsonTypeName;
+import io.druid.java.util.common.logger.Logger;
+import io.druid.security.basic.db.BasicSecurityStorageConnector;
+import io.druid.server.security.Access;
+import io.druid.server.security.Action;
+import io.druid.server.security.AuthenticationResult;
+import io.druid.server.security.Authorizer;
+import io.druid.server.security.Resource;
+import io.druid.server.security.ResourceAction;
+
+import java.util.List;
+import java.util.Map;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+@JsonTypeName("basic")
+public class BasicRoleBasedAuthorizer implements Authorizer
+{
+  private static final Logger log = new Logger(BasicRoleBasedAuthorizer.class);
+
+  private final BasicSecurityStorageConnector dbConnector;
+
+  @JsonProperty
+  private final boolean remapAuthNames;
+
+  @JsonCreator
+  public BasicRoleBasedAuthorizer(
+      @JacksonInject BasicSecurityStorageConnector dbConnector,
+      @JsonProperty("remapAuthNames") Boolean remapAuthNames
+  )
+  {
+    this.dbConnector = dbConnector;
+    this.remapAuthNames = remapAuthNames == null ? false : remapAuthNames;
+  }
+
+  @Override
+  public Access authorize(
+      AuthenticationResult authenticationResult, Resource resource, Action action
+  )
+  {
+    if (authenticationResult == null) {
+      return new Access(false);
+    }
+
+    String identity = null;
+    if (remapAuthNames) {
+      String authorizationName = dbConnector.getAuthorizationNameFromAuthenticationName(
+          authenticationResult.getIdentity()
+      );
+      if (authorizationName == null) {
+        return new Access(false);
+      } else {
+        identity = authorizationName;
+      }
+    } else {
+      identity = authenticationResult.getIdentity();
+    }
+
+    List<Map<String, Object>> permissions = dbConnector.getPermissionsForUser(identity);
+
+    // maybe optimize this later
+    for (Map<String, Object> permission : permissions) {
+      if (permissionCheck(resource, action, permission)) {
+        return new Access(true);
+      }
+    }
+
+    return new Access(false);
+  }
+
+  private boolean permissionCheck(Resource resource, Action action, Map<String, Object> permission)
+  {
+    ResourceAction permissionResourceAction = (ResourceAction) permission.get("resourceAction");
+    if (action != permissionResourceAction.getAction()) {
+      return false;
+    }
+
+    Resource permissionResource = permissionResourceAction.getResource();
+    if (permissionResource.getType() != resource.getType()) {
+      return false;
+    }
+
+    String permissionResourceName = permissionResource.getName();
+    Pattern resourceNamePattern = Pattern.compile(permissionResourceName);
+    Matcher resourceNameMatcher = resourceNamePattern.matcher(resource.getName());
+    return resourceNameMatcher.matches();
```

Comment:
Hmm, do you have any performance benchmark result? I'm not sure how this is fast enough.

Reply:
Not yet, I'll run some shortly